src/docfx/build/Build.cs

```diff
@@ -45,25 +45,36 @@ internal static class Build
             using (Progress.Start("Building files"))
             {
                 var sourceDependencies = new ConcurrentDictionary<Document, List<DependencyItem>>();
-                var fileListBuilder = new DocumentListBuilder();
+                var filesBuilder = new DocumentListBuilder();
+                var filesWithErrors = new ConcurrentBag<Document>();
```

Comment:
>ConcurrentBag [](start = 42, length = 13)

How about changing it to ConcurrentDictionary? ConcurrentBag might cause a memory leak here, since we only add items into it and use it for except, and never took them out afterwards. 
Please refer to [https://stackoverflow.com/questions/9256735/is-concurrentbag-cause-of-memory-leak](https://stackoverflow.com/questions/9256735/is-concurrentbag-cause-of-memory-leak)

Reply:
Interesting. I believe it should be collected unless the bag references itself directly or indirectly: [https://github.com/dotnet/corefx/issues/23068#issuecomment-321315063](https://github.com/dotnet/corefx/issues/23068#issuecomment-321315063)

Enumerating a ConcurrentBag does have a tiny performance penalty because internally it creates a snapshot for each enumeration: [https://github.com/dotnet/corefx/issues/2335](https://github.com/dotnet/corefx/issues/2335), we've already used linq here so the performance does not matter that much.


---
In reply to: [202224709](https://github.com/dotnet/docfx/pull/3064#discussion_r202224709) [](ancestors = 202224709)